moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -153,91 +163,96 @@ void JogCalcs::startMainLoop(JogArmShared& shared_variables, std::mutex& mutex)
     shared_variables.tf_moveit_to_cmd_frame = tf_moveit_to_cmd_frame_;
     mutex.unlock();
 
-    // Initialize the position filters to initial robot joints
-    while (!updateJoints() && ros::ok() && !stop_requested_)
+    // Only do jogging calculations if the robot should move, for efficiency
+    if (stop_requested_)
```

Comment:
Am I reading this wrong or does `stop_requested_` now only mean that the positions are reset, but the loop is kept alive? That's not what `stop_requested_` is supposed to do, we need an API to actually stop the loop as well. Stopping should be done by resetting the velocities in the command.

Reply:
The robot will not move if `stop_requested_` is true but the node will continue to make the end_effector_transform available (for closed loop controllers). It does look like the usage of `stop_requested_` needs updating in some places, though.

I think the way to exit from the loop is to destroy the object. Otherwise, we want it to continue to spin and provide the end effector transform (unless getEndEffectorTransform() should not be provided by this node.)

Over a year ago, the design decision was that you shouldn't need to publish zeros to stop motion. Because
a) ROS1 messages can be dropped
b) if you're spamming zeros to the controller, you shut down other ways of moving the robot. People may have multiple motion control modes, not just jogging. E.g. trajectory control and joint jogging possibly could operate with the same ros_control controller

Reply:
I guess this confusion was due to the naming of the variable. How about renaming:

`stop_requested_ -> exit_requested_`

To signify that it's not just meant to halt, but to exit.

Reply:
done